src/tracked_struct.rs

```diff
@@ -181,6 +186,60 @@ pub struct IdentityHash {
     hash: u64,
 }
 
+#[derive(Default, Debug)]
+pub(crate) struct IdentityMap {
+    // we use a non-hasher hashmap here as our key contains its own hash (in a sense)
```

Comment:
do you mind elaborating on what that sense is, or linking towards `Identity`? sorry, the internals Salsa can be a bit subtle, and I'm reviewing by campground rules 😄